src/solc_0.8/defi/rules/RequirementsRules.sol

```diff
@@ -0,0 +1,317 @@
+        _listERC721[newContract].maxAmountBalanceOf = maxAmountBalanceOf;
+        _listERC721[newContract].minAmountId = minAmountId;
+        _listERC721[newContract].maxAmountId = maxAmountId;
+        _listERC721[newContract].balanceOf = balanceOf;
+
+        // if it's a new member create a new registry, instead, only update
+        if (isERC721ListMember(newContract) == false) {
+            _listERC721Index.push(newContract);
+            _listERC721[newContract].index = _listERC721Index.length - 1;
+        }
+
+        emit ERC721ListRequirementSet(
+            contractERC721,
+            ids,
+            balanceOf,
+            minAmountBalanceOf,
+            maxAmountBalanceOf,
+            minAmountId,
+            maxAmountId
+        );
+    }
+
+    function setERC1155ListRequirement(
+        address contractERC1155,
+        uint256[] memory ids,
+        uint256 minAmountId,
+        uint256 maxAmountId
+    ) external onlyOwner isContract(contractERC1155) {
+        IERC1155 newContract = IERC1155(contractERC1155);
+        _listERC1155[newContract].ids = ids;
+        _listERC1155[newContract].minAmountId = minAmountId;
+        _listERC1155[newContract].maxAmountId = maxAmountId;
+
+        // if it's a new member create a new registry, instead, only update
+        if (isERC1155ListMember(newContract) == false) {
+            _listERC1155Index.push(newContract);
+            _listERC1155[newContract].index = _listERC1155Index.length - 1;
+        }
+
+        emit ERC11551ListRequirementSet(contractERC1155, ids, minAmountId, maxAmountId);
+    }
+
+    function getERC721ListRequirement(address contractERC721)
+        external
+        view
+        isContract(contractERC721)
+        returns (ERC721RequirementRule memory)
+    {
+        require(isERC721ListMember(IERC721(contractERC721)), "RequirementsRules: contract is not in the list");
+        return _listERC721[IERC721(contractERC721)];
+    }
+
+    function getERC1155ListRequirement(address contractERC1155)
+        external
+        view
+        isContract(contractERC1155)
+        returns (ERC1155RequirementRule memory)
+    {
+        require(isERC1155ListMember(IERC1155(contractERC1155)), "RequirementsRules: contract is not in the list");
+        return _listERC1155[IERC1155(contractERC1155)];
+    }
+
+    function deleteERC721ListRequirement(address contractERC721) external onlyOwner isContract(contractERC721) {
+        IERC721 reqContract = IERC721(contractERC721);
+        require(isERC721ListMember(reqContract), "RequirementsRules: contract is not in the list");
+        uint256 indexToDelete = _listERC721[reqContract].index;
+        IERC721 addrToMove = _listERC721Index[_listERC721Index.length - 1];
+        _listERC721Index[indexToDelete] = addrToMove;
+        _listERC721[addrToMove].index = indexToDelete;
+        _listERC721Index.pop();
+
+        emit ERC721RequirementDeleted(address(contractERC721));
```

Comment:
casting not needed

Reply:
done -> https://github.com/thesandboxgame/sandbox-smart-contracts/pull/605/commits/37c723ff5e4803acc897d63b6952df741381f358